src/swarm/neo/protocol/socket/MessageReceiver.d

```diff
@@ -90,6 +90,23 @@ private class MessageReceiverBase
 
     protected size_t buffer_content_end = 0;
 
+    /***************************************************************************
+
+        The maximum number of subsequent `read` calls without an I/O-wait call.
+
+    ***************************************************************************/
+
+    private const uint read_iowait_max = 10;
+
+    /***************************************************************************
+
+        Counts the maximum number of subsequent `read` calls without an I/O-wait
+        call.
+
+    ***************************************************************************/
+
+    private uint read_iowait_count;
```

Comment:
It could in principle be a local function variable, but it's not so easy. The loop which eventually calls `read()` repeatedly is in [`ConnectionBase.ReceiveLoop.fiberMethod`](https://github.com/sociomantic-tsunami/swarm/blob/v3.3.x/src/swarm/neo/connection/ConnectionBase.d#L560). I see two ways:
- Either the logic of suspending after `receiver.receive` has been called 10 times without calling the `cast(Event)this.suspend(...)` callback would have to go in this loop, relying on internal behaviour of `MessageReceiver`,
- or a counter method parameter would need to be added to `MessageReceiver.receive`, all the way down to where `read` is actually called.

Both ways look like a hassle to me.